<?xml version="1.0" ?>
<!-- This file was generated by robot assembler -->
<robot name="torso"
       xmlns:xi="http://www.w3.org/2001/XInclude"
       xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz"
       xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model"
       xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
       xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom"
       xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering"
       xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics"
       xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:macro name="gazebo_link_reference" params="link_name mu1:=0.9 mu2:=0.9" >
    <gazebo reference="${link_name}">
      <mu1>${mu1}</mu1>
      <mu2>${mu2}</mu2>
    </gazebo>
  </xacro:macro>
  <xacro:macro name="gazebo_transmission" params="joint_name" >
    <transmission name="${joint_name}_trans">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="${joint_name}">
        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
        <!--hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface-->
      </joint>
      <actuator name="${joint_name}_motor">
        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
        <!--hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface-->
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>
  </xacro:macro>

  <xacro:macro name="no_gazebo_transmission" params="joint_name" >
    <!-- dummy for fixed joint -->
  </xacro:macro>
  <xacro:gazebo_link_reference link_name="BASE" />
  <link name="BASE">
  <inertial>
     <mass value="0.402166"/>
     <origin rpy="0 0 0" xyz="3.505311e-17 -0.003352 0.021801"/>
     <inertia ixx="0.005514" ixy="-5.820766e-19" ixz="-5.820766e-20" iyy="0.005777" iyz="-1.675368e-05" izz="0.005409"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 0.000000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BATTERY-LEG-PLATE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.000000 0.000000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BATTERY-LEG-PLATE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.000000 0.016000" rpy="3.14159265 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BATTERY-BOX.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.000000 0.016000" rpy="3.14159265 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BATTERY-BOX.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.000000 0.025500 0.016000" rpy="-0.78539816 1.57079633 -2.35619449"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BODY-PLATE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 0.025500 0.016000" rpy="-0.78539816 1.57079633 -2.35619449"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BODY-PLATE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.033000 0.019500 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.033000 0.019500 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.033000 0.019500 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.033000 0.019500 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 0.019500 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.019500 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.025500 0.016000" rpy="1.38080804 1.57079633 2.95160437"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BODY-PLATE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.025500 0.016000" rpy="1.38080804 1.57079633 2.95160437"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BODY-PLATE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.033000 -0.019500 0.059500" rpy="1.57079633 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.033000 -0.019500 0.059500" rpy="1.57079633 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.046500 -0.007250 0.059500" rpy="-1.57079633 0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_ARM-SUPPORTER-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.046500 -0.007250 0.059500" rpy="-1.57079633 0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_ARM-SUPPORTER-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.033000 -0.007250 0.059500" rpy="-1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.033000 -0.007250 0.059500" rpy="-1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.033000 -0.019500 0.059500" rpy="1.57079633 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.033000 -0.019500 0.059500" rpy="1.57079633 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.046500 -0.007250 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_ARM-SUPPORTER-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.046500 -0.007250 0.059500" rpy="1.57079633 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_ARM-SUPPORTER-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.033000 -0.007250 0.059500" rpy="1.57079633 -0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.033000 -0.007250 0.059500" rpy="1.57079633 -0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.019500 0.059500" rpy="1.57079633 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.019500 0.059500" rpy="1.57079633 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.007250 0.073000" rpy="0.98279372 -1.57079633 0.58800260"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_ARM-SUPPORTER-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.007250 0.073000" rpy="0.98279372 -1.57079633 0.58800260"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_ARM-SUPPORTER-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.007250 0.059500" rpy="0.00000000 -0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.007250 0.059500" rpy="0.00000000 -0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.023500 -0.021000 -0.014000" rpy="-2.35619449 -1.57079633 0.78539816"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.023500 -0.021000 -0.014000" rpy="-2.35619449 -1.57079633 0.78539816"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.044500 -0.010500 -0.020000" rpy="-1.31911866 -1.57079633 1.31911866"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.044500 -0.010500 -0.020000" rpy="-1.31911866 -1.57079633 1.31911866"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.044500 -0.010500 -0.033000" rpy="-3.14159265 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.044500 -0.010500 -0.033000" rpy="-3.14159265 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.023500 -0.021000 -0.014000" rpy="-2.35619449 -1.57079633 0.78539816"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.023500 -0.021000 -0.014000" rpy="-2.35619449 -1.57079633 0.78539816"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.044500 -0.010500 -0.020000" rpy="1.87529262 -1.57079633 1.26630004"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.044500 -0.010500 -0.020000" rpy="1.87529262 -1.57079633 1.26630004"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-SPACER.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.044500 -0.010500 -0.033000" rpy="-3.14159265 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.044500 -0.010500 -0.033000" rpy="-3.14159265 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>
  <xacro:gazebo_link_reference link_name="LINK0" />
  <link name="LINK0">
  <inertial>
     <mass value="0.035533"/>
     <origin rpy="0 0 0" xyz="-5.150750e-18 1.149810e-17 0.025344"/>
     <inertia ixx="0.001006" ixy="-8.212501e-22" ixz="1.977899e-21" iyy="0.001006" iyz="-2.103206e-21" izz="0.001002"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 0.002500" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.000000 0.002500" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.000000 -0.013000 0.002500" rpy="1.57079633 -0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-FRAME-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 -0.013000 0.002500" rpy="1.57079633 -0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-FRAME-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.000000 0.000000 0.032500" rpy="1.10714872 1.57079633 -0.46364761"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 0.000000 0.032500" rpy="1.10714872 1.57079633 -0.46364761"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_SERVO-S3301.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 0.013000 0.002500" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-FRAME-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.013000 0.002500" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-FRAME-A.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>
  <xacro:gazebo_link_reference link_name="LINK1" />
  <link name="LINK1">
  <inertial>
     <mass value="0.01216"/>
     <origin rpy="0 0 0" xyz="0.014681 5.361282e-18 -0.017773"/>
     <inertia ixx="2.524786e-06" ixy="7.841632e-22" ixz="-7.542271e-08" iyy="3.418106e-06" iyz="1.027040e-21" izz="1.821371e-06"/>
  </inertial>
  <visual>
     <origin xyz="0.023000 0.000000 -0.004500" rpy="0.00000000 -1.57079633 -0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_UPPER-ARM26.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.023000 0.000000 -0.004500" rpy="0.00000000 -1.57079633 -0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_UPPER-ARM26.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.023000 0.000000 -0.017500" rpy="0.00000000 -1.57079633 -0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.023000 0.000000 -0.017500" rpy="0.00000000 -1.57079633 -0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_JOINT-BASE.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.023000 0.000000 -0.030500" rpy="0.76887886 1.57079633 -2.37271379"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-ARM28.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.023000 0.000000 -0.030500" rpy="0.76887886 1.57079633 -2.37271379"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/mesh_BOTTOM-ARM28.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>
  <xacro:gazebo_transmission joint_name="JOINT0" />
  <joint name="JOINT0" type="revolute">
     <parent link="BASE"/>
     <child link="LINK0"/>
     <origin xyz="0.000000 -0.007250 0.077500" rpy="-0.00000000 0.00000000 -1.57079633"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT1" />
  <joint name="JOINT1" type="revolute">
     <parent link="LINK0"/>
     <child link="LINK1"/>
     <origin xyz="-0.000000 -0.018000 0.032500" rpy="1.42371797 -1.57079633 0.14707836"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>

  <!-- Gazebo plugin for ROS Control -->
  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <!--robotNamespace>/</robotNamespace-->
      <legacyModeNS>false</legacyModeNS>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo>
</robot>
